File: scripts/serve-dist.mjs
import { createServer } from "node:http";
import { readFile, stat } from "node:fs/promises";
import path from "node:path";
import { fileURLToPath } from "node:url";

const ROOT = path.join(path.dirname(fileURLToPath(import.meta.url)), "..", "dist");
const PORT = process.env.PORT ? Number(process.env.PORT) : 8080;

const MIME = {
  ".html": "text/html; charset=utf-8",
  ".css": "text/css; charset=utf-8",
  ".js": "text/javascript; charset=utf-8",
  ".json": "application/json; charset=utf-8",
  ".png": "image/png",
  ".jpg": "image/jpeg",
  ".jpeg": "image/jpeg",
  ".webp": "image/webp",
  ".svg": "image/svg+xml",
  ".ico": "image/x-icon",
  ".woff": "font/woff",
  ".woff2": "font/woff2",
};

async function resolveFile(urlPath) {
  const decoded = decodeURIComponent(urlPath.split("?")[0]);
  let filePath = path.join(ROOT, decoded);
  if (!filePath.startsWith(ROOT)) {
    return null;
  }
  try {
    const stats = await stat(filePath);
    if (stats.isDirectory()) {
      filePath = path.join(filePath, "index.html");
    }
  } catch {
    return null;
  }
  try {
    await stat(filePath);
    return filePath;
  } catch {
    return null;
  }
}

createServer(async (req, res) => {
  const filePath = await resolveFile(req.url);
  if (!filePath) {
    res.writeHead(404, { "Content-Type": "text/plain; charset=utf-8" });
    res.end("404 Not Found");
    return;
  }
  const ext = path.extname(filePath).toLowerCase();
  const body = await readFile(filePath);
  res.writeHead(200, { "Content-Type": MIME[ext] || "application/octet-stream" });
  res.end(body);
}).listen(PORT, () => {
  console.log(`Serving dist/ at http://localhost:${PORT}`);
});

File: scripts/serve-dist.py
# Node が入っていない環境で dist/ を確認するための簡易サーバー。
# 通常は scripts/serve-dist.mjs を使う（こちらはフォールバック）。
import http.server
import os
import socketserver

DIST = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..", "dist")
PORT = int(os.environ.get("PORT", "8080"))

os.chdir(DIST)


class Handler(http.server.SimpleHTTPRequestHandler):
    def log_message(self, fmt, *args):
        print(fmt % args, flush=True)


with socketserver.TCPServer(("", PORT), Handler) as httpd:
    print(f"Serving dist/ at http://localhost:{PORT}", flush=True)
    httpd.serve_forever()
